src/jit/registerxmm.h

```diff
@@ -8,6 +8,10 @@
 #error  Must define REGDEF macro before including this file
 #endif
 
+#ifndef LEGACY_BACKEND
+#error This file is only used for the LEGACY_BACKEND build.
+#endif
+
```

Comment:
In general, I agree that we should delete any legacy AMD64 code as we find it. I didn't think about it in this case, but I'll try to remember it for next time.